### ISSUE TITLE
Unable to load (and save) context config file on Linux

### ISSUE THREAD

**Issue by chinatsu** [ggez/ggez#95]
Using x86_64 rustc 1.18.0 on Arch Linux, I don't seem to be able to load nor save configuration files. With some help from @icefoxen on IRC, I think we may have found out why this is.

The related bit of code I have been working on can be found here: https://github.com/chinatsu/ggetris/blob/master/src/main.rs#L54

I did some debug prints in my attempts to diagnose what was up, the comments represent what's printed from each of the prints.

```rust
println!("{:#?}", ctx.filesystem);
/* 
Filesystem {
    vfs: OverlayFS {
        roots: [
            <PhysicalFS root: /home/cn/Hello/dev/ggetris/target/debug/resources>,
            <PhysicalFS root: /home/cn/.local/share/ggetris>
        ]
    },
    resources_path: "/home/cn/Hello/dev/ggetris/target/debug/resources",
    zip_path: "/home/cn/Hello/dev/ggetris/target/debug/resources.zip",
    user_data_path: "/home/cn/.local/share/ggetris"
}
*/

println!("{:?}", ctx.filesystem.read_config());
// Err(ConfigError("Config file not found"))

println!("{:?}", ctx.filesystem.open("/conf.toml"));
// Ok(VfsFile)

println!("{:?}", ctx.filesystem.write_config(&conf::Conf::new()));
//  Err(ConfigError("Could not write config file because a directory is in the way?"))
```

As far as reading config files go, it seems that the context's filesystem sees the config file in the `user_data_path` with the prepending slash: `/conf.toml`. As such, `read_config()` doesn't match the `conf.toml` it's looking for. This is also why I can open the file by calling `ctx.filesystem.open("/conf.toml")`

I'm unsure what the `write_config` call tries to tell me. At first I thought maybe it tried to save the config in `target/debug/resources`, and that "resources" wasn't a directory. The error message remains unchanged after creating that directory. `~/.local/share/ggetris` is also a directory.

**Comment by icefoxen** [comment]
Yeah, this is because the "default config file" in filesystem.rs is "conf.toml", not "/conf.toml".  The way the file system code is set up, I had to basically either say "all paths must be absolute" or "all paths must be relative", and PhysicsFS chooses the former so I go along with that.  I just forgot to update the variable with the conf file name.  Easy fix.  This should probably be explained more fully in the documentation though.

More worryingly is the incorrect error message produced `write_config()`, which will take some research.  We really need better unit tests for this sort of thing, even if it's a PITA to try to set up a mocking system for it; probably better to have tests that leave test files hanging around than to not have tests at all.

**Comment by icefoxen** [comment]
Should be fixed by PR #96 🎶 Need to double-check and go through the code with a mop to make sure the error messages are all correct.

**Comment by icefoxen** [comment]
Ok, this appears to be fixed and I can't find anywhere else that obviously needs fixing.  We even have a unit test for this now, huzzah.  (We really need more of those.)
